### DISCUSSION TITLE
Enter key support in Combobox

### DISCUSSION THREAD

**Discussion by yhdesai** [tailwindlabs/headlessui#1123] [Ideas]
When you type some text in the combobox and it's only showing one result, You press enter directly without hovering or selecting that one result. The enter pressed should auto select the one result (assuming there's only one). If there's more than one result, It should select the first one in the list when enter is pressed.

**Comment by GregorShear** [comment]
I like this idea, and to add on a little bit, the first (or only) result should be "active" by default. Tailwind's Algolia-based command palette does what I'm looking for:

<img width="729" alt="Screen Shot 2022-02-27 at 8 37 20 PM" src="https://user-images.githubusercontent.com/3577568/155910637-60e1c709-42fb-478c-9fd1-67ec4d16505f.png">

**Reply by tan14142** [reply]
I have two api endpoints /autocomplete and /search. It would be nice if I could run a callback to fetch /search even if the options list from /autocomplete is empty or not loaded yet. Right now I am using onKeyUp={e => e.key === "Enter" && ...} on the input, which seems to work well.
